src/raft.c

```diff
@@ -599,6 +637,40 @@ static void clearClientSessions(RedisRaftCtx *rr)
     rr->client_session_dict = RedisModule_CreateDict(rr->ctx);
 }
 
+static void timeoutBlockedCommand(RedisRaftCtx *rr, raft_entry_t *entry)
+{
+    RedisModule_Assert(entry->data_len == sizeof(raft_index_t));
+    raft_index_t idx;
+    memcpy(&idx, entry->data, sizeof(idx));
+
+    BlockedCommand *bc = getBlockedCommand(idx);
+    if (!bc) {
+        /* unblock handler called before timeout was applied */
+        return;
+    }
+
+    if (RedisModule_CallReplyPromiseAbort(bc->reply, NULL) != REDISMODULE_OK) {
+        /* shouldn't happen with normal redis commands */
```

Comment:
it's part of the API, and it could happen in future with module based blocking commands (ala what we do in redisraft, wouldn't be able to abort us), but I think you're right in this case that it should be an assert.